lib/public_key/src/pubkey_cert.erl

```diff
@@ -144,14 +144,23 @@ validate_time(OtpCert, UserState, VerifyFun) ->
     {'Validity', NotBeforeStr, NotAfterStr} 
 	= TBSCert#'OTPTBSCertificate'.validity,
     Now = calendar:datetime_to_gregorian_seconds(calendar:universal_time()),
-    NotBefore = time_str_2_gregorian_sec(notBefore, NotBeforeStr),
-    NotAfter = time_str_2_gregorian_sec(notAfter, NotAfterStr),
 
-    case ((NotBefore =< Now) and (Now =< NotAfter)) of
-	true ->
-	    UserState;
-	false ->
-	    verify_fun(OtpCert, {bad_cert, cert_expired}, UserState, VerifyFun)
+    % Catch invalid date format errors in the `validity` field of the certificate
+    try
```

Comment:
This `try` statement seems wider in scope than it needs to be to me.  I think it would be better to isolate the parsing of `notBefore` and `notAfter` so the `try` statement doesn't also wrap the `verify_fun` call for `{bad_cert, cert_expired}`, which itself could raise a `function_clause` error.

Reply:
Changed into a clearer message.

Reply:
Reduced the scope of the try/catch and carried over the state.